lib/dispatcher/pool-base.js

```diff
@@ -65,6 +65,15 @@ class PoolBase extends DispatcherBase {
     }
 
     this[kOnConnectionError] = (origin, targets, err) => {
+      // If a connection error occurs, we remove the client from the pool,
+      // and emit a connectionError event. They will not be re-used.
+      // Fixes https://github.com/nodejs/undici/issues/3895
+      for (const target of targets) {
+        // Do not use kRemoveClient here, as it will close the client,
+        // but the client cannot be closed in this state.
+        this[kClients].splice(this[kClients].indexOf(target), 1)
+      }
```

Comment:
```suggestion
      this[kClients] = this[kClients].filter(target => !targets.includes(target))
```

Here's an alternative implementation that can handle the case when `target` isn't in the array.

Reply:
Thanks! I handled it without a filter.